src/DotLiquid/Strainer.cs

```diff
@@ -66,26 +87,34 @@ public Strainer(Context context)
             _context = context;
         }
 
+
         /// <summary>
         /// In this C# implementation, we can't use mixins. So we grab all the static
         /// methods from the specified type and use them instead.
         /// </summary>
         /// <param name="type"></param>
         public void Extend(Type type)
         {
-            // From what I can tell, calls to Extend should replace existing filters. So be it.
+            // From what I can tell, calls to Extend should replace existing filters with the same number of params. So be it.

```

Comment:
Probably a good time to revise the wording of this comment. "From what I can tell," and "So be it." does not add value but adds doubt in the implementation. Remove it assuming we are now asserting this is the behavior we specify.

Reply:
Fixed: https://github.com/dotliquid/dotliquid/pull/453/commits/09d0be13de6eac9119cb701eccb5c37ad36fa46f